server/src/test/java/org/elasticsearch/gateway/PriorityComparatorTests.java

```diff
@@ -106,75 +147,102 @@ protected Settings getIndexSettings(Index index) {
     public void testPriorityComparatorSort() {
         RoutingNodes.UnassignedShards shards = new RoutingNodes.UnassignedShards(mock(RoutingNodes.class));
         int numIndices = randomIntBetween(3, 99);
-        IndexMeta[] indices = new IndexMeta[numIndices];
-        final Map<String, IndexMeta> map = new HashMap<>();
+        IndexMetadata[] indices = new IndexMetadata[numIndices];
+        final Map<String, IndexMetadata> map = new HashMap<>();
 
         for (int i = 0; i < indices.length; i++) {
+            int priority = 0;
+            int creationDate = 0;
+            boolean isSystem = false;
+
             if (frequently()) {
-                indices[i] = new IndexMeta("idx_2015_04_" + String.format(Locale.ROOT, "%02d", i), randomIntBetween(1, 1000),
-                    randomIntBetween(1, 10000));
-            } else { // sometimes just use defaults
-                indices[i] = new IndexMeta("idx_2015_04_" +  String.format(Locale.ROOT, "%02d", i));
+                priority = randomIntBetween(1, 1000);
+                creationDate = randomIntBetween(1, 10000);
             }
-            map.put(indices[i].name, indices[i]);
+            if (rarely()) {
+                isSystem = true;
+            }
+            // else sometimes just use the defaults
+
+            indices[i] = IndexMetadata.builder("idx_2015_04_" + String.format(Locale.ROOT, "%02d", i))
```

Comment:
nit: since `i` can span to `99` using a prefix like "idx_" makes more sense

Reply:
Are you suggesting just `"idx_" + i`?

I'm not sure what the best thing to do here is. The comparator class just does a `compareTo` on the index names, so in a sense it doesn't matter what index names we use, so long as the result is sorted the way we expect. However, the *intention* of the sort is that newer indices, according to dates in the index names, are sorted first. So I wonder if it's worth preserving a date-like pattern in the index names for that reason? What do you think?

Reply:
I think that we should test what the code is effectively doing, ie alphabetically sorting by index names. Applying this to index names that contain a date is sort of a workaround as most of the time `SETTING_CREATION_DATE` should be present. Prioritizing index-0002 over index-0001 is also a thing.

Reply:
I swapped it for just `"idx_%04d"`. Any better?

Reply:
Thanks!